### PR TITLE
DOC: rewrite all build docs and restructure build/contributor guide

### LINKED ISSUE #15985: ENH, DOC: Add section explaining why and when to use a custom BLAS to build SciPy
### Is your feature request related to a problem? Please describe.

See https://github.com/scipy/scipy/pull/15947#discussion_r849163545

We have an FAQ item describing how to use a custom BLAS to build on Linux, but no explanation about why this is needed.

### Describe the solution you'd like.

Add an FAQ item or page on the contributing guide to explain this.

### Describe alternatives you've considered.

_No response_

### Additional context (e.g. screenshots, GIFs)

_No response_

### LINKED ISSUE #16322: DOC: building on Windows uses GCC with Meson, not MSVC
When looking at the meson-generated `build.ninja` on windows, on a machine that is duplicating the CI build, it seems there is no use of msvc only gcc. Is this correct? If so it might be nice to document this in the [meson build faq](https://scipy.github.io/devdocs/dev/contributor/meson.html#meson-frequently-asked-questions). Also I think the build requires the rtools gcc compiler, I have had problems with standard mingw.

Another niggle: `meson build --debug` on windows sets the gcc flags to `"-O2" "-g"`. Is using `-O2` intentional for the debug builds?

### LINKED ISSUE #17941: DOC: guidance on setting dev.py build -j flag in documentation, particularly if build hangs machine repeatedly
### Is your feature request related to a problem? Please describe.

When the amount of RAM per Ninja build task drops below a certain threshold, say, somewhere between 1-2 GB RAM, based on experience, the build system will wedge a machine because there's not enough memory available for all the compilation jobs it spawns [^0].  For example, on an `aarch64` machine with 2 Cortex-A76 cores and 6 Cortex-A55 cores with 8 GB RAM, the `dev.py` script uses the default number of build jobs, which is `n_logical_cpus + 2`, or 10 jobs. This setting reliably wedges that machine.

While overriding this setting for a `python dev.py build` invocation is relatively straightforward, after diagnosing the cause, I'm not sure how many people would get that far. It's also unclear to me how one would override that setting when installing a locally-built development version of `scipy`. I was using `python3 -m pip install -v --no-build-isolation` in my copy of the project root, based on the instructions in [How to build SciPy with Meson](https://docs.scipy.org/doc/scipy/dev/contributor/meson.html#meson-frequently-asked-questions), and couldn't figure out how I'd pass a `-j` setting through `pip` and the `meson-python` build backend to `meson` and `ninja`. (I finally hardcoded a value of 2 in `dev.py` just to get past this wedging issue.)  

[^0]: See, for instance, Gentoo's advice on `make -j`. <https://wiki.gentoo.org/wiki/MAKEOPTS>

### Describe the solution you'd like.

I don't think there's a great solution to this problem.

One path forward could be documentation, with some sort of section like, "If building `scipy` crashes your machine, run `ninja --help` on the command line after activating your virtual environment. In the output, there should be a line that starts with `-j`, and towards the end of that line, you'll see the default number of `ninja` jobs on your system. Divide the number of gigabytes of RAM installed on your system, e.g., 16 GB, by this default number of jobs. If this ratio is less than 2 GB RAM per job, when you next attempt to build `scipy` from source using the `dev.py` script, run `python dev.py build -j N`, where `N` is the largest natural number such that you still have at least 2 GB of RAM per job."

I don't know if I simply missed a better way to install a locally built development version of SciPy into my SciPy development environment, or if I'm just totally ignorant of the appropriate flags, but if there's a way to expose a `-j N` type of option whatever contributors are supposed to do that resembles `pip install -v --no-build-isolation .`, I could see that option being helpful.

Another path forward could be to try to set some kind of default maximum number of jobs, although I could see that sort of topic potentially becoming contentious, based on previous experience working on [Spack](https://github.com/spack/spack). If I remember correctly, most source-based package managers set a default max number of jobs; for example, [Spack](https://github.com/spack/spack/blob/2516ed181ad05c34a4eb0948ca4bc6bd567a99ba/lib/spack/spack/build_environment.py#L561) defaults the max number of build jobs to `max(num_logical_cpus, 16)`, and [Nix](https://nixos.org/manual/nix/unstable/command-ref/conf-file.html#conf-max-jobs) uses a conservative default maximum of 1 job. I don't have any preference, and would also support leaving the build settings as they are.

### Describe alternatives you've considered.

Doing nothing, I think, is a viable alternative, though I wonder how many other people have run into this issue; a cursory search of the project issues (e.g., `build crash`, `build hang`, `build lock`, `build wedge`, `hang`, `ninja hang`) did not yield much, aside from https://github.com/scipy/scipy/issues/10234.

Suggesting users cross-compile for particularly memory-constrained systems could be another viable option, or suggesting minimum development requirements.

### Additional context (e.g. screenshots, GIFs)

I'm sort of loath to deliberately crash my Chromebook again, but if need be, I can do it again, and take a screenshot because it runs Linux on a container. I have not tried to look at `crosh top` for system memory usage, plus I'm not sure how it use of ZRAM would affect any sort of memory contention; in general, the performance data seems a bit opaque to me. I was able to build `scipy` successfully, though, with `python dev.py build -j 4` and `python dev.py build -j 2`.

I was using ChromeOS 109 with a stock Debian 11 container, and all SciPy development system dependencies installed.

### LINKED ISSUE #18239: DOC: linking to custom BLAS/LAPACK locations is not clear
### Issue with current documentation:

I have tried to figure out how the `meson` (new) build system understands the linking stages for BLAS and LAPACK. But there seems to some missing points. Perhaps it is my ignorance in how `meson` understands flags etc.

So I want to build my own `scipy` and _NOT_ use `pip` or conda. I also want to forcefully go away from using distutils (to be prepared for 3.12).

**scipy version: 1.10.1 **

I have tried to understand this page: https://docs.scipy.org/doc/scipy/dev/contributor/building_faq.html#how-do-i-use-a-custom-blas-distribution-on-linux

However, when trying to build with meson:

```shell
scipy/meson.build:135:0: ERROR: Dependency "OpenBLAS" not found, tried pkgconfig and cmake
```

<details>

```shell
 meson setup build-dir -Dblas=blis
The Meson build system
Version: 1.0.1
Source dir: /home/software/bash-build/.compile/py-3.11/scipy-1.10.1
Build dir: /home/software/bash-build/.compile/py-3.11/scipy-1.10.1/build-dir
Build type: native build
Project name: SciPy
Project version: 1.10.1
C compiler for the host machine: gcc (gcc 12.2.0 "gcc (GCC) 12.2.0")
C linker for the host machine: gcc ld.bfd 2.40
C++ compiler for the host machine: g++ (gcc 12.2.0 "g++ (GCC) 12.2.0")
C++ linker for the host machine: g++ ld.bfd 2.40
Cython compiler for the host machine: cython (cython 0.29.33)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C supports arguments -Wno-unused-but-set-variable: YES 
Compiler for C supports arguments -Wno-unused-function: YES 
Compiler for C supports arguments -Wno-conversion: YES 
Compiler for C supports arguments -Wno-misleading-indentation: YES 
Compiler for C supports arguments -Wno-incompatible-pointer-types: YES 
Library m found: YES
Fortran compiler for the host machine: gfortran (gcc 12.2.0 "GNU Fortran (GCC) 12.2.0")
Fortran linker for the host machine: gfortran ld.bfd 2.40
Compiler for Fortran supports arguments -Wno-conversion: YES 
Checking if "-Wl,--version-script" : links: YES 
Program cython found: YES (/opt/gnu/12.2.0/python/packages/3.11.2/cython/0.29.33/bin/cython)
Program python3 found: YES (/opt/gnu/12.2.0/python/3.11.2/bin/python3)
Found pkg-config: /opt/generic/build-tools/1.0/bin/pkg-config (1.9.4)
Program pythran found: YES (/opt/gnu/12.2.0/python/packages/3.11.2/pythran/0.12.1/bin/pythran)
Run-time dependency threads found: YES
Library npymath found: YES
Library npyrandom found: YES
Run-time dependency blis found: YES 0.9.0
Found CMake: /opt/generic/build-tools/1.0/bin/cmake (3.25.3)
WARNING: CMake Toolchain: Failed to determine CMake compilers state
Run-time dependency openblas found: NO (tried pkgconfig and cmake)
Run-time dependency openblas found: NO (tried pkgconfig and cmake)

scipy/meson.build:135:0: ERROR: Dependency "OpenBLAS" not found, tried pkgconfig and cmake
```

</details>

basically regardless of the `blas=<>` argument. Tried, default, `none` and `blis` (my actual library).
The above is actually a bit misleading since users may think that this is the `blas` option which is not discovered, it is not, it is the lapack option.

I have `lapack` installed in a directory with no `pc` or cmake files. Is it possible to link that library to meson?
```shell
meson setup build-d -Dblas=blis -Dlapack=lapack
...
WARNING: CMake Toolchain: Failed to determine CMake compilers state
WARNING: CMake Toolchain: Failed to determine CMake compilers state
Run-time dependency lapack found: NO (tried pkgconfig and cmake)

scipy/meson.build:135:0: ERROR: Dependency "lapack" not found, tried pkgconfig and cmake
```

There is little documentation on how to bypass the dependency checking, and even with env-vars (not going to be supported: https://github.com/mesonbuild/meson/issues/9).
The whole problem of *fixing* names for dependencies makes it difficult to customize the build. I.e. if `-Dblas=<argument>` where `<argument>` has the restriction that it must be a name that used via `pkg-config`, i.e. `pkg-conf --libs <argument>`. So it seems to me that I have to provide a manual `*.pc` file for customizing a library usage? Really? :disappointed:   

I tried:
- put everything in FC flags etc., but this also fails since the dependency is forcefully added regardless of whether linking can occur
- put complete information in the `-Dlapack='-L/path/to/lapack/... -llapack` option, to no avail

## `python setup.py` works but

I can get scipy to be installed using the regular `setup.py` installation, but that will use my numpy `site.cfg` which I am not keen on.

### Idea or request for content:

It would be really valuable if the documentation showed how to link to a BLAS library located in a custom folder, *without* a `pc` file. That would show end-users how to play around with settings, and/or test various BLAS implementations (lets for now forget about flexiblas ;) ).

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by rgommers**
- Puts _Building from source as a top-level item_, with a coherent structure and the default (no customization) options for Windows/macOS/Linux and virtual/conda envs all on a single page with tabs 
- Adds a lot of info on how to customize builds and background content on Meson, linking out to other up-to-date docs where needed
- Removes all the duplicate and outdated sets of building docs across many pages
- Minor improvements to the contributor guide, e.g. using `dev.py` consistently and using `gh` to check out PRs
- Makes things more maintainable overall by avoiding writing on topics that inevitable will go out of date (e.g. the many tools for managing virtual envs)

Closes gh-18239 (by adding BLAS/LAPACK docs; also that issue is where this new structure was discussed)
Closes gh-16322 (by adding guidance on all supported Windows compilers)
Closes gh-17941 (by documenting how to add build customizations)
Closes gh-15985 (also BLAS/LAPACK related)

### CODE REVIEW

File: doc/source/building/compilers_and_options.rst
Comment (eli-schwartz) [inline doc/source/building/compilers_and_options.rst:27]: There is, technically, a warning_level=everything option, which provides a portable way to do clang -Weverything, MSVC /Wall, and GCC -Wlong -Wlist -Wof -Woptions (because GCC doesn't want to provide a convenient option to enable this).

File: doc/source/building/compilers_and_options.rst
Comment (eli-schwartz) [inline doc/source/building/compilers_and_options.rst:28]: Just --vsenv. The whole "pass a --flag option to another option" thing is part of the pip/build indirection layer. :D

File: doc/source/building/understanding_meson.rst
Comment (eli-schwartz) [inline doc/source/building/understanding_meson.rst:24]:
```suggestion
  build options, and create the build directory and ``build.ninja`` file,
```

File: doc/source/building/compilers_and_options.rst
Comment (rgommers) [reply doc/source/building/compilers_and_options.rst:28]: Thanks. I see that it can be `-Dvsenv=true` since Meson 1.1.0 (https://mesonbuild.com/Builtin-options.html#details-for-vsenv). That may be preferable for consistency?

Comment (rgommers) [reply doc/source/building/compilers_and_options.rst:28]: Had a look at the upstream change: https://github.com/mesonbuild/meson/pull/11620 says that `-Dvsenv=true` has more consistent behavior when reconfiguring, so I'll go with that. We already require Meson >=1.1.0, so it's fine from that perspective.

Comment (eli-schwartz) [reply doc/source/building/compilers_and_options.rst]: It doesn't have more consistent behavior when reconfiguring -- meson 1.1.0 has more consistent behavior when reconfiguring, because --vsenv is now saved as a builtin option.

On 1.1.0, --vsenv and -Dvsenv=true are parsed the same, and for the same reason that `--buildtype release` and `-Dbuildtype=release` are parsed the same.

Comment (rgommers) [reply doc/source/building/compilers_and_options.rst]: Ah, makes sense, thanks. Using `-D` still seems to make sense for consistency, given that we use `-Dbuildtype` and other `-D` options too.

**Comment by mdhaber** [comment]
Would you like comments on the Windows portion of this yet? I followed the instructions on a fresh machine and got MemoryErrors as in https://github.com/scipy/scipy/issues/17928#issuecomment-1533766621. Should I open a separate issue about that?

**Comment by rgommers** [comment]
> Would you like comments on the Windows portion of this yet? I followed the instructions on a fresh Windows machine and got `MemoryError`s as in [#17928 (comment)](https://github.com/scipy/scipy/issues/17928#issuecomment-1533766621). Should I open a separate issue about that?

gh-17928 is a very long issue about conda not resolving `environment.yaml`, so it'd be great if you could open a new issue about this MemoryError indeed (and Cc me). That's better than comments here, because it's probably some environment-related thing that won't really impact the content here beyond perhaps a warning.

**Comment by mdhaber** [comment]
Sure. There are separate (minor) comments about this that I'll add later.

**Review by tupui — COMMENTED** [review]
That's a nice rewrite 👏 

I am still not sure about removing the install page. Looking at the preview, we only get the install link as a small link alongside all other links. Looking at other projects in our realm, they all have some button or something more visible for install instruction.

Build instructions are quite prominent now, with a card on the main page and a link in the navbar. I don't think that basic users would be concerned about building SciPy (the devdoc page does not receive a lot of views in comparison to "actual API docs".) Hence, I would move the building part into the developer section.

In general, this section could be reworked/split or at least renamed because it has too much things from contributing, governance to roadmap.

**Comment by rgommers** [comment]
> Build instructions are quite prominent now, with a card on the main page and a link in the navbar. I don't think that basic users would be concerned about building SciPy (the devdoc page does not receive a lot of views in comparison to "actual API docs".) Hence, I would move the building part into the developer section.

We agreed on the current place in gh-18239 (including you, you gave it a thumbs up and I believe was also in the community call), and moving it now is a lot of work. So I don't really want to reconsider now to be honest.

> In general, this section could be reworked/split or at least renamed because it has too much things from contributing, governance to roadmap.

I agree in principle, but that's another significant amount of work, and out of scope for this PR. One thing at a time:)

> I am still not sure about removing the install page.

Nothing of interest is removed - our install page is https://scipy.org/install/.

**Comment by rgommers** [comment]
Oh okay, I get your comment now. The old top navbar was like this:

image

The new one is like this:

image

In both cases, the content is: the four tiles from the front page, plus _Release notes_. If it's changed to add a direct link to https://scipy.org/install/ named _Installing_ as the first item, I think that will address your comment. Right?

> During the call, we discussed having both the install instructions and building info on the same page. And we could use some selector (or our tab system) like PyTorch or scikit-learn to select a type of install or build.

This first sentence I said no to during the discussion. I think our install instructions should remain at https://scipy.org/install/. Either way, that is out of scope for this PR. The PyTorch-like selector widget would be great to have indeed.

**Comment by tupui** [comment]
> In both cases, the content is: the four tiles from the front page, plus Release notes. If it's changed to add a direct link to [scipy.org/install](https://scipy.org/install/) named Installing as the first item, I think that will address your comment. Right?

Yes that would if that's ok for all 😃 

For the rest, I agree we can iterate later when we work on the selector 👍

File: doc/source/building/index.rst
Comment (mdhaber) [inline doc/source/building/index.rst:154]: Is the Windows Universal C Runtime selected by default? In either case, it might be good to call out whether this is the *only* component that is needed so that the installation can be minimized, since a Visual Studio installation with lots of options can get quite large.

File: doc/source/building/index.rst
Comment (mdhaber) [inline doc/source/building/index.rst:148]: This sounds a bit daunting, but I was actually surprised at how easy it is now, at least compared to the past. 
1. Install Visual Studio
2. Install Chocolatey
3. Install `rtools` with Chocolatey (and make sure they're on the path)

Not really so bad!

File: doc/source/building/index.rst
Comment (mdhaber) [inline doc/source/building/index.rst:184]: Maybe link to Chocolatey installation instructions?

File: doc/source/building/index.rst
Comment (mdhaber) [inline doc/source/building/index.rst:283]: `source` is not a command on all Windows shells. I would find it helpful to note the equivalent command when `source` is not available. or maybe link to [How venvs work](https://docs.python.org/3/library/venv.html#how-venvs-work), which has a helpful table.

File: doc/source/building/index.rst
Comment (mdhaber) [inline doc/source/building/index.rst:303]: I might have a comment about this. This is a placeholder that I will edit or remove shortly.

File: doc/source/building/index.rst
Comment (mdhaber) [inline doc/source/building/index.rst:321]: IIUC, gh-18451 should make this reliable enough to recommend without `-jX`.👍

File: doc/source/building/index.rst
Comment (mdhaber) [inline doc/source/building/index.rst:143]: Is there a recommendation for those who don't already have compilers and who can't easily make an informed decision?

File: doc/source/dev/contributor/development_workflow.rst
Comment (mdhaber) [inline doc/source/dev/contributor/development_workflow.rst:112]:
```suggestion
environment, build, and run tests::
```
Doesn't look like this is an in-place build anymore, right?

**Review by mdhaber — COMMENTED** [review]
Now that I've run through these a few times, here are some suggestions that I would have found helpful.

Besides these, I think the bit about building on Windows in `dev_quickstart` can be updated, too. It currently links to [Building C and C++ Extensions](https://docs.python.org/3/extending/building.html#building), but now it can link to these instructions. Perhaps it is not essential, but for consistency, we could also update those to recommend `mamba` as these do.

Also, is there an easy way to do an in-place build with `dev.py`?

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst:154]: Yes, I believe it is - will add that.

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst:148]: > I was actually surprised at how easy it is now, at least compared to the past.

🎉. That is really good to hear, glad we're getting there also on Windows.

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst:283]: I like linking to the table, since this is a bit of a mess that I don't want to try to maintain in our own docs.

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst:303]: That must be new, I will check. It always defaulting to asking about adding to the path, and I have `$HOME/mambaforge/bin` as the very first entry on my PATH.

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst:321]: indeed.

**Comment by rgommers** [comment]
> Also, is there an easy way to do an in-place build with `dev.py`?

No. [meson-python now has editable installs](https://meson-python.readthedocs.io/en/latest/how-to-guides/editable-installs.html), but it's one or the other just like it always was for `runtests.py` vs. `setup.py develop`. I think we could have a go at unifying them at some point, but I don't trust editable installs enough just yet - they're for the early adopters right now I'd say.

File: doc/source/building/index.rst
Comment (mdhaber) [reply doc/source/building/index.rst:303]: It does give the option, but like Anaconda, the option it is uncheck by default and there is a note that recommends against it. (At least on Windows.)
I've always ignored the recommendation in the past and added it, but it might be worth a note.

File: doc/source/building/index.rst
Comment (mdhaber) [reply doc/source/building/index.rst:143]: That's what I would have said. Consider listing it first and noting that it's recommended if the reader is unsure.

File: doc/source/building/index.rst
Comment (mdhaber) [reply doc/source/building/index.rst:148]: Yeah, I would even suggest that this paragraph can be removed.

File: doc/source/building/index.rst
Comment (mdhaber) [reply doc/source/building/index.rst:184]: Also I just tried `choco install rtools` instead of `choco install rtools -y --no-progress --force --version=4.0.0.20220206` and I was not successful in building. Maybe pin the version?

Comment (andyfaff) [reply doc/source/building/index.rst:184]: The latest version of rtools available via choco places the build tools in a different location to that which we use in 4.0.0.20220206. I've not looked into where they actually go with the new version. I think that the newer version uses gcc12, the older version gcc10

Comment (andyfaff) [reply doc/source/building/index.rst:184]: Also, choco can be used to install pkg-config.

Comment (mdhaber) [reply doc/source/building/index.rst:184]: Yes, there were two locations with `gfortran.exe`: `rtools43\x86_64-w64-mingw32.static.posix\bin` and `rtools43\usr\lib\mxe\usr\x86_64-w64-mingw32.static.posix\bin`. I chose the one with the shorter name.

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst:148]: I am toning it down a bit, but it should be kept I'd say - it still is a multi-step prep process that can not be fully described in the docs, and may go wrong. Even if it got better, the contrast with Linux/macOS where all you need is
```
mamba env create -f environment.yaml
python dev.py build
```
is significant.

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst:184]: Is it `choco install pkgconfiglite`? I guess that is needed only when not using conda.

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst:303]: I can't figure it out, there are no docs for Windows it seems, beyond the one-liner at https://github.com/conda-forge/miniforge#windows. @mdhaber would you mind opening an issue on that repo? I don't have a Windows machine and not enough context to open an issue there or respond to questions.

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst:143]: > That's what I would have said. Consider listing it first and noting that it's recommended if the reader is unsure.

Agreed, doing this.

> The one potential hangup I see is putting it on the path. There is a note about the necessity of doing that, but can we mention the specific location or recommend how to look for it?

Mingw-w64 is already on the path if you use `choco`, and there's the recommendation for how to check in the `.. note::`: run `gfortran --version`.

Comment (rgommers) [reply doc/source/building/index.rst:143]: This is the problem with Windows: too many manual steps, shells with different behavior, etc. I don't think it can be described in more detail without it becoming a maintenance burden again, and it's impossible to be completely exhaustive. Windows users who struggle with these instructions are better off using WSL, then things are trivially easy.

File: doc/source/building/index.rst
Comment (mdhaber) [reply doc/source/building/index.rst:312]: Opened conda-forge/miniforge#453.

Comment (rgommers) [reply doc/source/building/index.rst:312]: Thanks Matt. First response there seems to indicate it may be a bug. Either way, I think there is nothing to do here - if it stays like that and conda-forge adds docs, we can point to them.

**Comment by rgommers** [comment]
This is ready again, all comments are addressed I believe. I resolved the straightforward ones, and left some of Matt's Windows ones open for double checking.

File: doc/source/building/index.rst
Comment (mdhaber) [reply doc/source/building/index.rst]: I'll leave this open for @andyfaff, but it's resolved from my perspective. I don't know about pkg-config.

File: doc/source/building/index.rst
Comment (mdhaber) [reply doc/source/building/index.rst]: > Mingw-w64 is already on the path

I'll have to check on that. It might be visible from Powershell but I'm not sure if it is for other shells.

I agree that it's impossible to be exhaustive and keep it up to date. If it is not added to the path for all shells by default, the one easy thing we might do is mention that the default location of the compilers can be found in "SciPy GitHub Actions CI jobs for Windows", which is already linked.

Comment (mdhaber) [reply doc/source/building/index.rst]: Thanks for reordering these. The one thing I'd suggest is to also reorder the tabs so that the MinGW-w64 instructions show by default.

File: doc/source/dev/dev_quickstart.rst
Comment (mdhaber) [inline doc/source/dev/dev_quickstart.rst:92]: The "With [system-level dependencies](https://output.circle-artifacts.com/output/job/84fc2e60-7fad-4280-b709-3f164d02db9a/artifacts/0/html/building/index.html#system-level) installed" bit only appears in the Virtual env tab. I think it is needed in the Conda env tab, too (for Windows at least).

**Review by mdhaber — APPROVED** [review]
I've suggested a few other modifications, but overall the Windows build instructions LGTM.

File: doc/source/dev/dev_quickstart.rst
Comment (rgommers) [reply doc/source/dev/dev_quickstart.rst:53]: Good point. I think we should always put Conda first, because it's easier to work with and more suitable than pip/venv overall.

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst]: It's only needed when not using conda indeed, I'll add a comment behind the command.

File: doc/source/building/index.rst
Comment (rgommers) [reply doc/source/building/index.rst]: Good idea. I reordered that tab as well now, resolving.

File: doc/source/dev/dev_quickstart.rst
Comment (rgommers) [reply doc/source/dev/dev_quickstart.rst:92]: Made the text here identical now, and because of that could move it outside the tabs now. Resolving.

File: doc/source/building/index.rst
Comment (andyfaff) [inline doc/source/building/index.rst:169]: I think pkg-config might be provided by rtools. At least it doesn't need an extra install step in the CI config.

File: doc/source/building/index.rst
Comment (andyfaff) [reply doc/source/building/index.rst]: In the windows CI pkg-config is already installed. I'm not sure if it's part of rtools. Pkgconfiglite may be the one.

Comment (rgommers) [reply doc/source/building/index.rst]: From the Rtools description in Chocolatey:

_Rtools is a toolchain bundle used for building R packages from source (those that need compilation of C/C++ or Fortran code) and for build R itself. Rtools43 is used for R-devel, the development version of R since revision 83535 and is intended for R 4.3.0._

_Rtools43 consists of Msys2 build tools, GCC 12/MinGW-w64 compiler toolchain, libraries built using the toolchain, and QPDF. Rtools43 supports 64-bit Windows and UCRT as the C runtime._

_Compared to Rtools42, Rtools43 has newer versions of three core components: GCC, MinGW-w64, and binutils. It is therefore recommended to re-compile all code with the new toolchain to avoid problems. The code compiled by even earlier versions of Rtools is incompatible due to use of MSVCRT and has to be recompiled with Rtools43 for use in R packages._

So that's not completely clearcut - `pkg-config` is not part of `binutils`. I downloaded the built sources as a zipfile from https://cran.r-project.org/bin/windows/Rtools/rtools43/files/, and in `installed.list` we find `pkgconf 1.8.0`. So yes, that's it - it is included. Hence I can remove this one line again.

Comment (rgommers) [reply doc/source/building/index.rst]: I believe that's the conclusive answer, so I'll resolve this.

**Comment by rgommers** [comment]
This has an approval from @mdhaber, and @tupui seemed happy as well. I'm pretty keen to get this merged, both because it's quite useful as a reference, and because doc rebuilds after branch switches are super slow. Does anyone have further comments that are important to address before merging?

**Review by tupui — APPROVED** [review]

**Comment by rgommers** [comment]
Another approval in and no more comments, so I'll hit the green button here. Thanks a lot for the thorough testing Matt, and the review Pamphile.

**Pull request merged**
